components/page-header/page-header.component.ts

```diff
@@ -91,9 +97,10 @@ const NZ_CONFIG_MODULE_NAME: NzConfigKey = 'pageHeader';
     '[class.ant-page-header-compact]': 'compact',
     '[class.ant-page-header-rtl]': `dir === 'rtl'`
   },
-  imports: [NzOutletModule, NzIconModule]
+  imports: [NzOutletModule, NzIconModule, NzButtonModule]
 })
 export class NzPageHeaderComponent implements AfterViewInit, OnDestroy, OnInit {
+  private location = inject(Location, { optional: true });
```

Comment:
```suggestion
  private location = inject(Location);
```

Reply:
If we do that we will have breaking change, maybe people use this component without using the routing and in this case, Location provider will not be found resulting to an injector error

Reply:
As far as I know, Location comes from @angular/common and it is provided in the root.

Reply:
you are right: https://github.com/angular/angular/blob/main/packages/common/src/location/location.ts

let me implement your changes :)